client/src/components/ConsoleUsersListGroup.js

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import authorizeUser from '../Auth'
+import { ButtonGroup, Button, Toast, ToastBody, ToastHeader, 
+  Row, Col, Alert, Collapse } from 'reactstrap';
+  import Loading from './Loading'
+
+const ConsoleUsersList = props => {
+  const [errMsg, setErrMsg] = useState("");
+
+  const ApproveRequest = (user_id, type) => {
+    console.log('Approve', user_id, props.group_id)
+    let token = props.token;
+    console.log(token)
+
+    let apiendpoint = "/group"
+    if (type === 'accept'){
+      apiendpoint += "/accept_user_request/" + props.group_id;
+    } else {
+      apiendpoint += "/deny_user_request/" + props.group_id;
+    }
+
+    console.log(props.group_id, apiendpoint)
+    authorizeUser(token, apiendpoint, {'user_id':user_id}, 'patch')
+      .then(result => {
+        console.log("result request to join:",result)
+        if (result){
+          console.log(result)
+          window.location.reload(false);
+        }
+        else {
+          console.log('Error')
+          setErrMsg("An error has occurred. Please refresh and try again.")
+        }
+      })
+      .catch(error => {
+        console.log(error)
+        setErrMsg("An error has occurred. Please refresh and try again.")
+      })
+  }
+
+  return(
+    Object.keys(props.users).reverse().map(function(key) {

```

Comment:
what is the purpose of reverse() here? i feel like if we want a specific order, then we should set that order in the backend when it returns the response

Reply:
Reverse reverses the list so latest results come first (also present in posts, notifications). I agree it makes more sense to do it in the backend, but mongoose is strange and needs what seems to be an shoehorned workaround for push to front. So when I first encountered this issue I decided to just reverse it client side, but this will eventually have potentially serious performance effects. We need to look into and test MongoDB push to front and implement it for all 3 (maybe one more? idr) cases.

Reply:
discussed offline, found we can we use `.sort({"created": -1})` as a way to order results in the backend when we make mongoose queries.